test/compiler/inference.jl

```diff
@@ -1388,16 +1386,16 @@ gg13183(x::X...) where {X} = (_false13183 ? gg13183(x, x) : 0)
 
 # test the external OptimizationState constructor
 let linfo = get_linfo(Base.convert, Tuple{Type{Int64}, Int32}),
-    world = typemax(UInt),
+    world = UInt(23)
```

Comment:
23??

Reply:
If you have defined fewer than 22 methods, test this could fail, but I hoped it was a safe assumption.

Reply:
Why doesn't `get_world_counter()` work here? At the very least this probably deserves a comment.

Reply:
it would also work, indeed any number `1:get_world_counter()` is expected to be valid

Reply:
Either 1 or `get_world_counter()` seems better then.

Reply:
Those are the valid values for the constructor, but would weaken the test (we know the extrema must work from everywhere else, here we want to test the interior region). For the test below to be especially helpful it should be >= 3 and <= primary-world(method) - 2 < world-counter. The value 23 seemed like a perfectly reasonable number within those constraints.